sdk/core/azure-core/azure/core/pipeline/policies/_redirect.py

```diff
@@ -152,12 +163,19 @@ def send(self, request):
         """
         retryable = True
         redirect_settings = self.configure_redirects(request.context.options)
+        original_domain = get_domain(request.http_request.url) if redirect_settings["allow"] else None
         while retryable:
             response = self.next.send(request)
             redirect_location = self.get_redirect_location(response)
             if redirect_location and redirect_settings["allow"]:
                 retryable = self.increment(redirect_settings, response, redirect_location)
                 request.http_request = response.http_request
+                if domain_changed(original_domain, request.http_request.url):
```

Comment:
Should client authors have an option to ignore this and retain the authorization header on redirects? I think the communications service is one example where they expect redirects in some scenarios and also require those to still be authorized.